File: backend/app/schemas/comment.py
from datetime import datetime

from pydantic import BaseModel, Field, field_validator

from app.schemas.user import UserBrief


class CommentCreateRequest(BaseModel):
    content: str = Field(
        ...,
        description="评论内容，不能为空，去除首尾空格后最长 500 字",
        examples=["这条帖子很赞！"],
    )

    @field_validator("content")
    @classmethod
    def validate_content(cls, v):
        v = v.strip()
        if not v:
            raise ValueError("评论内容不能为空")
        if len(v) > 500:
            raise ValueError("评论最多500字")
        return v


class CommentResponse(BaseModel):
    id: int
    post_id: int
    content: str
    author: UserBrief
    created_at: datetime


CommentResponse.model_rebuild()

File: frontend/app/src/main/java/com/rdbb/jiancircle/data/model/Comment.kt
package com.rdbb.jiancircle.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** 评论，对应后端 schemas/comment.py 的 CommentResponse */
@Serializable
data class Comment(
    val id: Int = 0,
    @SerialName("post_id")
    val postId: Int = 0,
    val content: String = "",
    val author: User = User(id = 0, username = "", nickname = "未知用户"),
    @SerialName("created_at")
    val createdAt: String = ""
)

/** 发表评论请求，对应后端 CommentCreateRequest */
@Serializable
data class CommentRequest(
    val content: String
)
